src/envoy/http/service_control/filter.cc

```diff
@@ -35,14 +45,57 @@ Envoy::Http::FilterHeadersStatus ServiceControlFilter::decodeHeaders(
     Envoy::Http::RequestHeaderMap& headers, bool) {
   ENVOY_LOG(debug, "Called ServiceControl Filter : {}", __func__);
 
-  Envoy::Tracing::Span& parent_span = decoder_callbacks_->activeSpan();
+  if (!headers.Method()) {
```

Comment:
Do you want to add these here already? I'd prefer to add these in when we actually remove path matcher.

If we add these in this PR, we should also copy in unit tests that cover these cases.

Reply:
Well, I just moved these checking and x-http-method-override here too.   I have added more unit-tests for them